pkg/cli/render.go

```diff
@@ -27,12 +28,21 @@ type Render struct {
 }
 
 func (s *Render) Run(cmd *cobra.Command, args []string) error {
-	c, err := s.client.CreateDefault()
+	var c client2.Client
+
+	imageAndArgs := imagesource.NewImageSource(s.File, args, s.Profile, nil, false)
+
+	_, file, err := imageAndArgs.ResolveImageAndFile()
 	if err != nil {
 		return err
 	}
-
-	imageAndArgs := imagesource.NewImageSource(s.File, args, s.Profile, nil, false)
+	if file == "" {
+		// Lazily create client so that local file render doesn't require an API connection
+		c, err = s.client.CreateDefault()
```

Comment:
I may be mistaken, but isn't this what the [DeferredClient](https://github.com/acorn-io/runtime/blob/5e6684245d93e7e2b256698da3fbacd90d3ed0bc/pkg/client/deferred.go#L13) is for?